test/libc-testsuite/sscanf.c

```diff
@@ -212,6 +212,12 @@ static int test_sscanf(void)
         /* legacy stdio fails this test */
 	TEST(i, sscanf("hello, world\n", "%8c%8c", a, b), 1, "%d fields, expected %d");
 	TEST_S(a, "hello, wX", "");
+
+    a[0] = '#';
+    a[1] = '\0';
+    TEST(i, sscanf(" Nan(98b)", "%lf%c", &d, a), 1, "got %d fields, expected %d");
+    TEST(i, isnan(d), 1, "isnan %d expected %d");
+    TEST_S(a, "#", "");
```

Comment:
Please add tests for the failing cases too, at least missing closing paren, invalid character inside parens and overrunning a field width value. And make sure `inf` and `infinity` don't accidentally consume a following (valid) paren sequence nor fail to convert on an invalid paren sequence.

Reply:
Done.

Reply:
Thanks. The tests look good. I'll let you iterate until they all pass CI and check back.